tests/tester.nim

```diff
@@ -42,5 +39,60 @@ proc testWsConflict() =
   removeDir("tests/ws_conflict/dpkg")
   removeDir(myproject)
 
+type
+  Node = object
+    name: string
+    versions: seq[string]
+    deps: seq[string]
+
+proc createNode(name: string, versions: seq[string], deps: seq[string]): Node =
+  result = Node(name: name, versions: versions, deps: deps)
+
+template withDir(dir: string; body: untyped) =
+  let old = getCurrentDir()
+  try:
+    setCurrentDir(dir)
+    body
+  finally:
+    setCurrentDir(old)
+
+proc createNimblePackage(node: Node) =
+  let packagePath = node.name
+  createDir(packagePath)
+  withDir packagePath:
+    var nimbleContent = &"""
+# Nimble package file for {node.name}
+author = "Your Name"
+license = "MIT"
+
+# Dependencies
+    """
+    for d in node.deps:
+      nimbleContent.add &"requires \"{d}\""
+    writeFile(node.name & ".nimble", nimbleContent)
+
+proc testSemVer() =
+  # Example graph data
+  var graph: seq[Node] = @[
+    createNode("A", @["1.0.0", "1.1.0", "2.0.0"], @[]),
+    createNode("B", @["2.1.0", "3.0.0", "3.1.0"], @["A >= 1.0.0"]),
+    createNode("C", @["1.2.0", "1.2.1"], @["B >= 2.0.0"]),
+    createNode("D", @["1.0.0", "1.1.0", "1.1.1"], @[]),
+    createNode("E", @["2.0.0", "2.0.1", "2.1.0"], @["D >= 1.0.0"]),
+    createNode("F", @["1.0.0", "1.0.1", "1.1.0"], @["E >= 2.0.0"]),
+  ]
+
+  createDir "source"
+  withDir "source":
+    for i in 0..<graph.len:
+      createNimblePackage graph[i]
+
+  createDir "myproject"
+  withDir "myproject":
+    exec atlasExe & " --showGraph use F"
```

Comment:
`atlasExe` is a relative path, so it doesn't work here, probably use absoulte path when outputing the binary.